### ISSUE TITLE
React Content Query WebPart : Fields with spaces in their internal names not working

### ISSUE THREAD

**Issue by spplante** [SharePoint/sp-dev-fx-webparts#316]
#### Category
- [ ] Question
- [x] Bug
- [ ] Enhancement

#### Expected or Desired Behavior
When using the React Content Query WebPart, displaying a field that has spaces in its internal name should render the field value just like a normal field.

#### Observed Behavior
When using the React Content Query WebPart, displaying a field that has spaces in its internal name (which are automatically replaced with "_x0020_" by SharePoint), the value is of the field is displayed as blank.

#### Steps to Reproduce
- Create a field called "MyTestField"
- Create a field called "My Test Field"
- Select both fields using the Content Query WebPart
- The field using spaces should render a blank value

**Comment by spplante** [comment]
It looks like either the REST api (responsible for executing the CAML query) or Handlebars (the templating engine) doesn't like the underscore in the internal name. I will take a look at this issue tonight and submit a pull request accordingly.

**Comment by spplante** [comment]
Bug is now fixed, added the fix to pull request #313 that should soon be merged. 

The bug was due to a really odd behavior that the REST API does with fields that have spaces in their internal names. As we know, a field created with the display name **My Field** will generate the internal name **My_x0020_Field** since SharePoint replaces spaces by `_x0020_`. So far this is all good, but it gets uglier when executing a CAML query using the REST API. For that same field, the API will let you get the field value using the matching internal name **My_00x20_Field** which is expected, but for getting the _htmlValue_ and _textValue_, you have to use the following name : **My_x005f_x0020_x005f_Field**

So the REST API basically replaces all occurences of **_00x20_** by **_x005f_x0020_x005f_**. I fixed it by changing line **549** of **ContentQueryService.ts** to the following :

```typescript
let spacesFormattedName = viewField.replace(new RegExp("_x0020_", "g"), "_x005f_x0020_x005f_");

normalizedResult[viewField] = {
    textValue: result.FieldValuesAsText[spacesFormattedName],
    htmlValue: result.FieldValuesAsHtml[spacesFormattedName],
    rawValue: result[viewField] || result[viewField + 'Id']
};
```

@gurleen23. you can now wait for pull request #313 to be merged, or you can download the latest version on the **dev** branch of my [fork](https://github.com/spplante/sp-dev-fx-webparts/tree/dev/samples/react-content-query-webpart) if you don't want to wait. Thanks for the input!

@VesaJuvonen, you can close this issue as soon as pull request #313 is merged. Thanks!

**Comment by spplante** [comment]
Forgot to close this issue. Fixed since PR #313.
